### PR TITLE
RR-395 - Added PrisonerSearchSummary

### PR DESCRIPTION
**Pull request by nathanrussell-moj-digital**
This PR adds the `location` field to `PrisonerSummary` because we can populate that field from the `Prisoner` record that we get back from `prisoner-search-api`
It also adds `PrisonerSearchSummary` which extends `PrisonerSummary` and adds fields that will be populated from calls made to the PLP and CIAG APIs.

### CODE REVIEW

File: integration_tests/mockData/prisonerByIdData.ts [hunk 1]
```diff
@@ -8,11 +8,15 @@ const prisoners = {
...
       headers: { 'Content-Type': 'application/json;charset=UTF-8' },
       jsonBody: {
+        prisonId: 'MDI',
         prisonerNumber: 'A00001A',
         firstName: 'Paris',
         lastName: 'Jones',
-        releaseDate: new Date().toISOString(),
+        releaseDate: '2030-12-31',
         nonDtoReleaseDateType: 'HDC',
+        receptionDate: '2001-17-01',
```
Comment (nathanrussell-moj-digital) [inline integration_tests/mockData/prisonerByIdData.ts:17]: Our wiremock stubs in our integration tests were missing a few fields from `prisoner-search-api`

**Review by chrimesdev — APPROVED** [review]

**Comment by chrimesdev** [comment]
LGTM

File: server/@types/viewModels/index.d.ts [hunk 2]
```diff
@@ -7,6 +7,12 @@ declare module 'viewModels' {
     lastName: string
     receptionDate?: Date
     dateOfBirth?: Date
+    location: string
+  }
+
+  export interface PrisonerSearchSummary extends PrisonerSummary {
+    hasCiagInduction: boolean
+    hasActionPlan: boolean
   }
```
Comment (mattwills-moj-digital) [inline server/@types/viewModels/index.d.ts:16]: 👍

**Review by mattwills-moj-digital — APPROVED** [review]
LGTM

**Pull request merged**